pinot-core/src/main/java/org/apache/pinot/core/segment/index/readers/ImmutableDictionaryReader.java

```diff
@@ -244,6 +252,6 @@ protected String getPaddedString(int dictId, byte[] buffer) {
   }
 
   protected byte[] getBuffer() {
-    return new byte[_numBytesPerValue];
+    return (_valueReader instanceof VarLengthBytesValueReaderWriter) ? null : new byte[_numBytesPerValue];
```

Comment:
Is there a way to avoid 'instanceof'?

Reply:
If we want to avoid it, it's easily doable. Given that `_valueReader` is a final field, we can have a boolean for this condition and use that here.